common/component/redis/redis.go

```diff
@@ -187,11 +210,135 @@ func ParseClientFromProperties(properties map[string]string, componentType metad
+				tokenErr := kitretry.NotifyRecover(
+					func() error {
+						var innerTokenErr error
+						token, innerTokenErr = (*cred).GetToken(ctx, policy.TokenRequestOptions{
+							Scopes: []string{"https://redis.azure.com/.default"},
+						})
+						return innerTokenErr
+					},
+					backoffManager,
+					func(err error, _ time.Duration) {
+						(*logger).Debugf("redis client: entraID token acquisition failed with error: %v. Retrying...", err)
+					},
+					func() {
+						(*logger).Debug("redis client: entraID token acquisition succeeded after error")
+					},
+				)
+				if tokenErr != nil {
+					_ = client.Close()
+					(*logger).Fatalf("redis client: entraID token acquisition failed: %v", tokenErr)
+					return
+				}
+
+				// Use the new access token via the Redis AUTH command
+				backoffManager = backoffConfig.NewBackOffWithContext(ctx)
+				authErr := kitretry.NotifyRecover(
+					func() error {
+						var innerAuthErr error
+						innerAuthErr = client.AuthACL(ctx, username, token.Token)
+						return innerAuthErr
+					},
+					backoffManager,
+					func(err error, _ time.Duration) {
+						(*logger).Debugf("redis client: entraID auth failed with error: %v. Retrying...", err)
+					},
+					func() {
+						(*logger).Debug("redis client: entraID auth succeeded after error")
+					},
+				)
+				if authErr != nil {
+					_ = client.Close()
+					(*logger).Fatalf("redis client: entraID auth failed: %v", authErr)
+					return
+				}
+				// Since the entraID auth succeeded we are setting the duration to wait for the next iteration of the refresh loop
+
+				(*logger).Debugf("redis client: entraID auth token successfully refreshed with the server")
+
+				tokenRefreshDuration = time.Until(token.ExpiresOn.Add(-refreshGracePeriod))
+			}
+		}
+	}(cred, username, logger)
+}
+
+func (s *Settings) GetEntraIDCredentialAndSetInitialTokenAsPassword(ctx context.Context, properties *map[string]string) (*time.Time, *azcore.TokenCredential, error) {
+	if len(s.Password) > 0 || len(s.Username) > 0 {
+		return nil, nil, errors.New(
+			"redis client configuration error: username or password must not be specified when using Entra ID authentication")
+	}
+	envSettings, envErr := azure.NewEnvironmentSettings(*properties)
+	if envErr != nil {
+		return nil, nil, fmt.Errorf("redis client configuration error: %w", envErr)
+	}
+	cred, tokenCredErr := envSettings.GetTokenCredential()
+	if tokenCredErr != nil {
+		return nil, nil, fmt.Errorf("redis client configuration error: %w", tokenCredErr)
+	}
+
+	token, tokenErr := cred.GetToken(ctx, policy.TokenRequestOptions{
+		Scopes: []string{"https://redis.azure.com/.default"},
+	})
+	if tokenErr != nil {
+		return nil, nil, fmt.Errorf("redis client configuration error: %w", tokenErr)
+	}
+
+	s.Password = token.Token
+
+	type CustomClaims struct {
+		Oid string `json:"oid"`
+		jwt.RegisteredClaims
+	}
+
+	parsedToken, _, parseErr := jwt.NewParser().ParseUnverified(token.Token, &CustomClaims{})
```

Comment:
Nit: please add a comment explaining that we use ParseUnverified because the Azure SDK already validated the token